src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -823,6 +823,8 @@ private bool WriteLocalFileHeader(bool isEmptyFile)
                 }
                 else // if we are not in streaming mode, we have to decide if we want to write zip64 headers
                 {
+                    // When updating and when we can seek, the data descriptor bit needs to be flipped back to prevent corruption
```

Comment:
` When updating and when we can seek` or ` When updating or when we can seek`? the `if` condition is `and` suggesting this shoudl be `or` unless there is some logic elsewhere that prevents one without hte other.

Reply:
Thanks. I will change the comment to your suggestion:

```suggestion
                    // We are in seekable mode so we will not need to write a data descriptor
```